service/storage_service.cc

```diff
@@ -425,6 +451,10 @@ future<> storage_service::sync_raft_topology_nodes(mutable_token_metadata_ptr tm
             if (!prev_normal.contains(id)) {
                 co_await notify_joined(*ip);
             }
+
+            if (const auto it = host_id_to_ip_map.find(host_id); it != host_id_to_ip_map.end() && it->second != *ip) {
```

Comment:
I don't see any code that would add / update new mapping in this map, if one appears

Reply:
So it looks like we'll purge the old IP only in very specific scenario, the first time we change a node's IP... Then when it's changed again into a third IP, we won't find the second, so we won't purge it

Reply:
`host_id_to_ip_map` is local to the `sync_raft_topology_nodes` invocation, each node is processed only once

Reply:
it's a local multishard read from memory (it should reside in row_cache), how  can it be more expensive than e.g. `o_dsync` write?

Reply:
Anyway the simplest way would be to run the 100 node again and see whether there's any significant regression, maybe my worries are unfounded, @gusev-p could you please check?

Reply:
We can address in follow-up if it turns out it's a problem.

Reply:
I ran it (still getting 1000-sec timeout):
`INFO  2024-02-15 17:37:43,406 [shard 0:strm] raft_topology - sync_raft_topology_nodes, total us 289570, transition us 0, left us 0, total_update_peer_info_us 1158, total_update_peer_info_count 60, total_normal_us 1838, total_normal_count 61 get_host_id_to_ip_map_us 373 total_sys_ks_us 287646`

Reply:
The previous result was (https://github.com/scylladb/scylladb/issues/17039#issuecomment-1935634888)
```
INFO  2024-02-09 10:58:14,679 [shard 0:strm] raft_topology - sync_raft_topology_nodes, total us 169330, transition us 76, left us 0, total_normal_us 1072, total_normal_count 57
```

so there is some regression, although better what we started with

(or maybe it's variability)